### ISSUE TITLE
Some usernames toss an error when getting the logs

### ISSUE THREAD

**Issue by kristopolous** [regisb/slack-cli#5]
I mentioned this [in this reddit comment](https://www.reddit.com/r/Python/comments/6z1jt3/slack_cli_for_productive_developers/dn2h3cw/) which I'm essentially copying here:

In `names.cli:25`, if a person's account gets deleted then the lookup of the name by id fails and raises a `slacker.Error: user_not_found`. A possible way to deal with this would be something like
```
    try:
        self.user_index[user_id] = slack.client().users.info(user_id).body['user']['name']
    except:
        self.user_index[user_id] = str(user_id)
```
There's no truly awesome way to go about it.

**Comment by regisb** [comment]
Hi @kristopolous,
Thanks for taking the time to open this issue! I am not observing the same behaviour for a deleted user. When I make a call to `users.info` with an existing but deleted user's ID, I do get a valid response. That response includes a `"deleted": True` entry. This is compliant with the Slack API documentation: https://api.slack.com/methods/users.info

Have you seen this behaviour on a real channel? Are you sure that it was for a deleted user?

**Comment by kristopolous** [comment]
Thanks it's not a deleted user actually. This is a real bug. It may be due to bots or events or something like that. Let me demonstrate it:

Without the patch I'm going to try to log this channel which has a lot of automated scripts posting in it:

```
$ slack-cli -s fleet --last 100 > fleet.log 
Traceback (most recent call last):
  File "/home/chris/.local/bin//slack-cli", line 11, in <module>
    sys.exit(main())
  File "/home/chris/.local/lib/python3.5/site-packages/slackcli/cli.py", line 10, in main
    sys.exit(run())
  File "/home/chris/.local/lib/python3.5/site-packages/slackcli/cli.py", line 47, in run
    last_messages(args.src, args.last)
  File "/home/chris/.local/lib/python3.5/site-packages/slackcli/cli.py", line 87, in last_messages
    utils.search_messages(source, count=count)
  File "/home/chris/.local/lib/python3.5/site-packages/slackcli/utils.py", line 82, in search_messages
    print(format_message(source_name, message))
  File "/home/chris/.local/lib/python3.5/site-packages/slackcli/utils.py", line 88, in format_message
    names.username(message['user']), message['text']
  File "/home/chris/.local/lib/python3.5/site-packages/slackcli/names.py", line 36, in username
    return UserIndex.instance().name(user_id)
  File "/home/chris/.local/lib/python3.5/site-packages/slackcli/names.py", line 26, in name
    self.user_index[user_id] = slack.client().users.info(user_id).body['user']['name']
  File "/home/chris/.local/lib/python3.5/site-packages/slacker/__init__.py", line 161, in info
    return self.get('users.info', params={'user': user})
  File "/home/chris/.local/lib/python3.5/site-packages/slacker/__init__.py", line 90, in get
    api, **kwargs
  File "/home/chris/.local/lib/python3.5/site-packages/slacker/__init__.py", line 72, in _request
    raise Error(response.error)
slacker.Error: user_not_found
```
Alright that's cool, let's find out what's going on

The last line that doesn't get included is from the Jibble app. Furthermore, if you look for type(user_id) you get the empty string "", the line `print("(", type(user_id), len(user_id), user_id == '', ")")` shows `( <class 'str'> 0 True )`

So then we are entering this function _without_ a user-id. This is coming from https://github.com/regisb/slack-cli/blob/master/slackcli/utils.py#L78  and an example of an offending packet, through `print(message)` looks like this:

```
{'text': '*** *jibbled out* via Slack ', 'team': '***', 'next': {...} , 'type': 'message', 'channel': {...}, 'permalink': '***', 'previous': {...}, 'next_2': {...}, 'username': 'jibble', 'iid': '***', 'ts': '1505372476.000032', 'user': ''}
```
Aha, so our `user` is empty but our `username` is there. So maybe if we change things to this:

>  name = names.username(message['user']) if ('user' in message and len(message['user']) > 0) else message['username']

And then hope things are ok. 

I've put the changes in a PR that you'll be receiving shortly

**Comment by regisb** [comment]
Again, thanks a lot for taking the time to investigate this issue! I think you nailed it. I commented on your PR.

**Comment by regisb** [comment]
Thanks @kristopolous for your help! I think I did manage to fix this in commit 178d996 please feel free to reopen if it's not the case.
